pkg/notification/notification.go

```diff
@@ -111,6 +118,18 @@ func (es *UnlimitedEventSource) Run() {
 	es.doneChan <- struct{}{}
 }
 
+func (es *UnlimitedEventSource) AddInterval(interval time.Duration) error {
```

Comment:
Basically this function can be read as `DisableInterval`
any other option return an error :)

Reply:
Enables the interval just once, and only if the value is >= 0 :)

Reply:
I'm torn if this should be renamed `SetInterval` to emphasize that we can have 0-1 timers in the current implementation :\

Reply:
> I'm torn if this should be renamed SetInterval to emphasize that we can have 0-1 timers in the current implementation :\

Sounds more accurate than `AddInterval`